### PR TITLE
Problem: `cosmos-sdk` and `ibc-go` version not upgraded to latest

### LINKED ISSUE #708: Problem: `cosmos-sdk` and `ibc-go` version not upgraded to latest
Latest released Cosmos SDK version: `0.45.1`
Latest released IBC version: `3.0.0-rc0`

Checklist:

- [ ] Upgrade `cosmos-sdk` to `v0.45.1`
- [ ] Upgrade `ibc-go` to `v3.0.0-rc0`
- [ ] Make necessary code changes and fix integration tests
- [ ] Update `gomod2nix.toml`
- [ ] Update `cosmos-sdk` git submodule
- [ ] Add upgrade handler
- [ ] Add upgrade test
- [ ] Add changelog

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Upgraded `cosmos-sdk` and `ibc-go` versions. Fixes #708.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -459,30 +467,23 @@ func New(
 	app.SetAnteHandler(anteHandler)
 	app.SetEndBlocker(app.EndBlocker)
 
-	planName := "v3.0.0"
-	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, plan upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
+	planName := "v4.0.0"
+	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
 		// a new param in 1.0.0 -- set to roundup(5x 7secs) as a safe choice
 		app.IBCKeeper.ConnectionKeeper.SetParams(ctx, ibcconnectiontypes.NewParams(uint64(40*time.Second)))
```
Comment (tomtau) [inline app/app.go:473]: this doesn't need to be set any more in the new upgrade handler?

File: app/app.go [hunk 2]
```diff
@@ -459,30 +467,23 @@ func New(
...
-		fromVM := make(map[string]uint64)
-		for moduleName := range app.mm.Modules {
-			fromVM[moduleName] = 1
-		}
-		// delete new modules from the map, for _new_ modules as to not skip InitGenesis
-		delete(fromVM, authz.ModuleName)
-		delete(fromVM, feegrant.ModuleName)
+
```
Comment (tomtau) [inline app/app.go:474]: should add interchain accounts?
https://github.com/cosmos/ibc-go/blob/v3.0.0-rc1/docs/migrations/v2-to-v3.md#upgrade-proposal

File: go.mod [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+module github.com/crypto-org-chain/chain-main/v4
 
 go 1.17
 
 require (
 	github.com/confluentinc/bincover v0.1.0
-	github.com/cosmos/cosmos-sdk v0.44.6
-	github.com/cosmos/ibc-go/v2 v2.0.3
+	github.com/cosmos/cosmos-sdk v0.45.1
+	github.com/cosmos/ibc-go/v3 v3.0.0-rc0
```
Comment (tomtau) [inline go.mod:8]: rc1 is the latest now?

File: app/app.go (hunk 2 above)
Comment (devashishdxt) [reply app/app.go:474]: Created an issue for this: https://github.com/crypto-org-chain/chain-main/issues/719
Will do it in future PR.

File: go.mod (hunk 3 above)
Comment (devashishdxt) [reply go.mod]: Changed

File: app/app.go (hunk 1 above)
Comment (devashishdxt) [reply app/app.go]: Removed

**Review by tomtau — APPROVED** [review]
ok if the interchain accounts are added in a separate PR

File: integration_tests/test_staking.py [hunk 4]
```diff
@@ -105,12 +106,26 @@ def test_staking_redelegate(cluster):
     assert rsp["code"] == 0, rsp["raw_log"]
     delegation_info = cluster.get_delegated_amount(signer1_address)
     old_output = delegation_info["delegation_responses"][0]["balance"]["amount"]
-    cluster.redelegate_amount(
-        validator1_operator_address,
-        validator2_operator_address,
-        "2basecro",
-        signer1_address,
+    cli = cluster.cosmos_cli()
```
Comment (yihuang) [inline integration_tests/test_staking.py:109]: we should vendor the cli wrappers later I think, in another PR.

**Review by yihuang — APPROVED** [review]
LGTM

**Pull request merged**